test/ctr.bats

```diff
@@ -996,6 +996,136 @@ function check_oci_annotation() {
 	crictl exec --sync "$ctr_id" findmnt -no TARGET,PROPAGATION "$CTR_DIR" | grep -v private
 }
 
+@test "ctr that mounts container storage as read-only option but not recursively" {
+	# See https://www.shellcheck.net/wiki/SC2154 for more details.
+	declare stderr
+
+	# Parent of "--root", keep in sync with test/helpers.bash file.
+	TEST_VOLUME="$TESTDIR"/test-volume
+
+	mkdir -p "$TEST_VOLUME"
+	mount -t tmpfs none "$TEST_VOLUME"
+
+	mkdir -p "$TEST_VOLUME"/test-sub-volume
+	mount -t tmpfs none "$TEST_VOLUME"/test-sub-volume
+
+	PARENT_DIR="$TEST_VOLUME"
+	CTR_DIR="/host"
+
+	jq --arg path "$PARENT_DIR" --arg ctr_dir "$CTR_DIR" \
+		'  .mounts = [ {
+			host_path: $path,
+			container_path: $ctr_dir,
+			readonly: true,
+			propagation: 0
+		} ]' \
+		"$TESTDATA"/container_sleep.json > "$TESTDIR"/config
+
+	start_crio
+
+	ctr_id=$(crictl run "$TESTDIR"/config "$TESTDATA"/sandbox_config.json)
+
+	run ! --separate-stderr crictl exec --sync "$ctr_id" touch /host/test
+	[[ "$stderr" == *"Read-only file system"* ]]
+
+	crictl exec --sync "$ctr_id" touch /host/test-sub-volume/test
+}
```

Comment:
@littlejawa, do you happen to know off the cuff, perhaps, whether any read-only mounts under Kata are also recursively read-only for the entire tree? So, would a sub-volume mount also be read-only?

I am asking, as this simple above test does not pass under Kata, whereas it has no issues on a vanilla tux container.

Reply:
Probably not. I am getting the same failure on Fedora.